dlt/destinations/snowflake/configuration.py

```diff
@@ -0,0 +1,74 @@
+from typing import Final, Optional, Any, Dict
+
+from sqlalchemy.engine import URL
+
+from dlt.common.typing import TSecretStrValue
+from dlt.common.configuration.specs import ConnectionStringCredentials
+from dlt.common.configuration import configspec
+from dlt.common.destination.reference import DestinationClientDwhConfiguration
+
+
+def _read_private_key(private_key: str, password: Optional[str] = None) -> bytes:
+    """Load an encrypted or unencrypted private key from string.
+    """
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives.asymmetric import rsa
+    from cryptography.hazmat.primitives.asymmetric import dsa
+    from cryptography.hazmat.primitives import serialization
+
+    pkey = serialization.load_pem_private_key(
+        private_key.encode(), password.encode() if password is not None else None, backend=default_backend()
+    )
+    return pkey.private_bytes(
+        encoding=serialization.Encoding.DER,
+        format=serialization.PrivateFormat.PKCS8,
+        encryption_algorithm=serialization.NoEncryption()
+    )
+
+
+@configspec
+class SnowflakeCredentials(ConnectionStringCredentials):
+    drivername: Final[str] = "snowflake"  # type: ignore[misc]
+    password: Optional[TSecretStrValue] = None
+    host: str = None
+    database: str = None
+    warehouse: Optional[str] = None
+    role: Optional[str] = None
+    private_key: Optional[TSecretStrValue] = None
+    private_key_passphrase: Optional[TSecretStrValue] = None
```

Comment:
please add `password`, `warehouse` and `role` like `location` for `GcpCredentials`:
```
__config_gen_annotations__: ClassVar[List[str]] = ["location"]
```
why we do that: so when the secrets are auto generated for the user s/he sees those even if they are optionsl